cmd/up.go

```diff
@@ -290,26 +301,36 @@ func (up *UpContext) devMode(isRetry bool) error {
 		d = deployments.GevDevSandbox(up.Dev)
 		create = true
 	}
-	progress := newProgressBar("Activating your Okteto Environment...")
-	progress.start()
-	defer progress.stop()
 
 	if isRetry && !deployments.IsDevModeOn(d) {
 		return fmt.Errorf("Okteto Environment has been deactivated")
 	}
+	deploys[up.Dev.Name] = d
 
-	for i := range up.Dev.Volumes {
-		if err := volumes.Create(up.Context, up.Dev.GetDataVolumeName(i), up.Dev, up.Client); err != nil {
+	for _, s := range up.Dev.Services {
+		if _, ok := deploys[s.Name]; ok {
+			continue
+		}
+		d, err := deployments.Get(s.Name, up.Dev.Namespace, up.Client)
+		if err != nil {
 			return err
 		}
+		deploys[s.Name] = d
 	}
 
-	c, err := deployments.DevModeOn(d, up.Dev, create, up.Client)
-	if err != nil {
+	if err := deployments.TraslateDevMode(deploys, up.Dev); err != nil {
 		return err
 	}
 
-	up.Container = c.Name
+	for name, d := range deploys {
+		forceCreate := false
+		if d.Name == name {
```

Comment:
what is this check for?

Reply:
it is to handle de autocreation of the main service. for the other ones, for now, they must be already created

Reply:
could we have that separate from the loop so it's more obvious? Or maybe have a function called "isMainService" so it's easier to troubleshoot in the future?